programmers/신규-아이디-추천.js

```diff
@@ -0,0 +1,21 @@
+function solution(new_id) {
+    new_id = new_id
+        .toLowerCase()
+        .replace(/[^\w\.\-]/g, '')
+        .replace(/[\.]{2,}/g, '.')
+        .replace(/^\./, '')
+        .replace(/\.$/, '');
+    
+    if (!new_id)  new_id = "a";
+    if (new_id.length >= 16) {
+        new_id = new_id
+            .slice(0, 15)
+            .replace(/\.$/, '');
+    }
+    if (new_id.length <= 2) {
+      while (new_id.length < 3) {
+        new_id = new_id + new_id[new_id.length -1];
+     } 
+   } 
+   return new_id;
+}
```

Comment:
a = a + b 같은 방식일 경우 a += b로 작성 가능
```suggestion
    if (new_id.length <= 2) {
      while (new_id.length < 3) {
        new_id += new_id[new_id.length -1];
      } 
    } 
   return new_id;
}
```

Reply:
마지막 if문을 이렇게도 표현 할 수 있음
- padEnd() : 현재 문자열에 다른 문자열(new_id[new_id.length - 1])을 채워, 주어진 길이(3)를 만족하는 새로운 문자열을 반환
```js
return new_id.length > 2 ? new_id : new_id.padEnd(3, new_id[new_id.length - 1]);
```